maintenance/delete_workflows.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+set -e
+
+# Simple script to remove workflows that are not successful or failed (with completion status like skipped, cancelled, etc.)
+# It also removes successful or failed workflows that doesn't have artifacts or logs.
+# You need to be logged to 'gh' first
+
+# owner/repo syntax
+repo="$1"
+
+temp_file="workflow_runs.payload"
+rm -rf $temp_file
+
+echo "Fetching workflows..."
+echo $(gh api /repos/${repo}/actions/runs --paginate | jq '{count: .total_count, runs: [.workflow_runs[] | select(.status=="completed") | {id, conclusion}]}') | jq '.' > $temp_file
+
+ids_to_delete=$(cat "$temp_file" | jq -r '.runs | .[] | select((.conclusion!="success") and (.conclusion!="failure")) | .id')
+
+if [ "${ids_to_delete}" = "" ]
+then
+	echo "All workflows are exited successfully or failed. Nothing to remove"
+else
+	echo "Removing all the workflows that are not successful or failed..."
+	while read -r line; do
+		id="$line"
+		## Workaround for https://github.com/cli/cli/issues/4286 and https://github.com/cli/cli/issues/3937
+		echo -n | gh api --method DELETE /repos/${repo}/actions/runs/${id} --input -
+		
+		echo "Stale workflow run with ID $id deleted successfully!"
+	done <<< $ids_to_delete
+fi
+
+ids_to_delete=$(cat "$temp_file" | jq -r '.runs | .[] | select((.conclusion=="success") or (.conclusion=="failure")) | .id')
+
+if [ "${ids_to_delete}" = "" ]
+then
+	echo "No workflows to check for logs or artifacts. Exiting..."
+else
+	echo -e "\nDeleting workflows without logs and artifacts..."
+	while read -r line; do
+		id="$line"
+		artifact_count=$(gh api /repos/${repo}/actions/runs/${id}/artifacts | jq -r '.total_count')
+		if [ "${artifact_count}" = "0" ]
```

Comment:
No, artifacts != logs. Artifacts are the binaries produced by workflows and uploaded using [this action](https://github.com/actions/upload-artifact). More info [here](https://docs.github.com/es/actions/guides/storing-workflow-data-as-artifacts).

Right now we don't use artifacts anywhere but in JMP. See [this workflow as an example](https://github.com/jellyfin/jellyfin-media-player/actions/runs/1087841247). Artifacts are located at the bottom. [Logs are the output of the job](https://github.com/jellyfin/jellyfin-media-player/runs/3214843031?check_suite_focus=true)

[Artifacts also expire](https://docs.github.com/es/actions/guides/storing-workflow-data-as-artifacts#configuring-a-custom-artifact-retention-period), but might have a different expiration date than logs. That's why we need to set a preference order on what's more important, logs or artifacts.

For my script's logic, I choose that **artifacts > logs**. I assume that maintainers/teams that have configured the artifact retention period to be larger than the retention for logs have their reasons to do so, so I'm not allowed to remove workflows runs that might have valuable data.

## How the script works:

As I mentioned, for each workflow, the script checks if there are artifacts:

* **IF TRUE (total_count != 0)**: We continue the loop to the next one and ignore if the workflow run has logs or not because artifacts are more relevant.
* **IF FALSE (total_count == 0)**: We call the API endpoint to download the logs (there is no API endpoint for querying for its existence, so I must perform the download straightaway). If the logs doesn't exist, GitHub will reply with **HTTP 401 Gone**, so we can remove the workflow run without further ado.

gh exits with code ``1`` when it receives the ``401``, so we can use the standard ``||`` in bash to run the deletion command. This happens in this line: 

```
gh api --silent /repos/${repo}/actions/runs/${id}/logs || \ 
				echo -n | gh api --method DELETE /repos/${repo}/actions/runs/${id} --input - && \
				echo "Workflow run without logs and artifacts with ID $id deleted successfully!"
```

Reply:
wow I must have been blind not seeing that `gh api --silent /repos/${repo}/actions/runs/${id}/logs`, all good then xD